kotlin/browse-ogc-api-feature-service/src/main/java/com/esri/arcgisruntime/sample/browseogcapifeatureservice/MainActivity.kt

```diff
@@ -0,0 +1,246 @@
+            val featureCollectionTitleAdapter = ArrayAdapter(
+                this,
+                R.layout.feature_collection_title_list_item,
+                featureCollectionTitles
+            )
+
+            featureCollectionTitleListView.apply {
+                // add the adapter to the list view
+                adapter = featureCollectionTitleAdapter
+                // set an on item click listener for items in the list view
+                setOnItemClickListener { _, _, position, _ ->
+                    // load the selected collection
+                    loadLayer(featureCollectionInfos[position])
+                }
+            }
+
+            // expand the layers list
+            layerFAB.isExpanded = true
+        }
+    }
+
+    /**
+     * Load  and query features from the given OGC Feature Collection. Create a Feature Layer from
+     * the table and add it to the map's operational layers.
+     */
+    private fun loadLayer(selectedCollectionInfo: OgcFeatureCollectionInfo) {
+        // create the OGC feature collection table
+        val table = OgcFeatureCollectionTable(selectedCollectionInfo)
+
+        // populate the OGC feature collection table
+        val queryParameters = QueryParameters().apply {
+            maxFeatures = 1000
+        }
+
+        table.apply {
+            // set the feature request mode to manual (only manual is currently supported). In this mode,
+            // you must manually populate the table - panning and zooming won't request features
+            // automatically
+            featureRequestMode = ServiceFeatureTable.FeatureRequestMode.MANUAL_CACHE
+
+            // populate the table from the service
+            populateFromServiceAsync(queryParameters, false, null)
+
+            addDoneLoadingListener {
+                // create a feature layer from the OGC feature collection table
+                val ogcFeatureLayer = FeatureLayer(table).apply {
+                    // Choose a renderer for the layer based on the table.
+                    renderer = getRendererForTable(table) ?: this.renderer
+                }
+
+                mapView.map.operationalLayers.apply {
+                    // clear previous layers from the map
+                    clear()
+                    // add the layer to the map
+                    add(ogcFeatureLayer)
+                }
+
+                // zoom to the extent of the selected collection
+                val collectionExtent = selectedCollectionInfo.extent
+                if (!collectionExtent.isEmpty) {
+                    mapView.setViewpointGeometryAsync(collectionExtent, 100.0)
+                }
+
+                // hide the layer list
+                layerFAB.isExpanded = false
+            }
+        }
+    }
+
+    /**
+     * Return a simple renderer for points, lines and polygons.
+     */
+    private fun getRendererForTable(table: FeatureTable): Renderer? {
```

Comment:
similar to above, this could be clearer using `featureTable` as the variable name

```suggestion
    private fun getRendererForTable(featureTable: FeatureTable): Renderer? {
```